AddressSpace/templates/designToSourceVariablesBody.jinja

```diff
@@ -0,0 +1,411 @@
+{
+
+/* The thread pool should be initialized by Meta while reading the config file, using function: 
+    SourceVariables_initSourceVariablesThreadPool */
+static Quasar::ThreadPool *sourceVariableThreads = nullptr;
+void SourceVariables_initSourceVariablesThreadPool (unsigned int minThreads, unsigned int maxThreads, unsigned int maxJobs)
+{
+  LOG(Log::DBG) << "Initializing source variables thread pool to min=" << minThreads  << " max=" << maxThreads << " threads maxJobs=" << " jobs";
+  sourceVariableThreads = new Quasar::ThreadPool (maxThreads, maxJobs);
+}
+
+void SourceVariables_destroySourceVariablesThreadPool ()
+{
+  if (sourceVariableThreads)
+  {
+    delete sourceVariableThreads;
+    sourceVariableThreads = nullptr;
+  }
+}
+
+Quasar::ThreadPool* SourceVariables_getThreadPool () { return sourceVariableThreads; }
+}
+
+#ifndef BACKEND_OPEN62541
+
+
+#include <iomanager.h>
+
+#include <iostream>
+#include <stdexcept>
+
+#include <QuasarThreadPool.h>
+  
+{% for className in designInspector.get_names_of_all_classes() %}
+  {% if designInspector.objectify_source_variables(className)|length > 0 %}
+    #include <AS{{className}}.h>
+    {% if designInspector.class_has_device_logic(className) %}
+      #include <D{{className}}.h>
+    {% endif %}
+  {% endif %}
+{% endfor %}
+
+namespace AddressSpace
+{
+
+{% for className in designInspector.get_names_of_all_classes(only_with_device_logic=True) %}
+  {% for sv in designInspector.objectify_source_variables(className, restrict_by="[@addressSpaceRead='asynchronous' or @addressSpaceRead='synchronous']") %}
+    class IoJob_{{className}}_READ_{{sv.get('name')}} : public Quasar::ThreadPoolJob
+    {
+      public:
+        IoJob_{{className}}_READ_{{sv.get('name')}} (
+          IOManagerCallback *callback,
+          OpcUa_UInt32 hTransaction,
+          OpcUa_UInt32 callbackHandle,
+          const UaNode* parentObjectNode
+        ):
+          m_callback(callback),
+          m_hTransaction(hTransaction),
+          m_callbackHandle (callbackHandle),
+          m_parentObjectNode (parentObjectNode)
+        {}
+        
+      virtual void execute ()
+      {
+        LOG(Log::DBG) << 
+          "Executing IoJob read: className={{className}} varName={{sv.get('name')}}" <<
+          " hTransaction:" << m_hTransaction << 
+          " cbkhandle " << m_callbackHandle;
+        UaStatus s;
+        {{sv.get('dataType')}} value;
+        UaDateTime sourceTime;
+        // Obtain Device Logic object
+        const AS{{className}}* addressSpaceObject (nullptr);
+        addressSpaceObject = dynamic_cast<const AS{{className}}*> ( m_parentObjectNode );
+        if (addressSpaceObject == m_parentObjectNode)
+        { /* OK. Proper cast. */
+          Device::D{{className}}* device = addressSpaceObject->getDeviceLink();
+          if (device != 0)
+          {
+            /* OK - Device Logic linked. */
+            {% if sv.get('addressSpaceReadUseMutex') == 'of_this_operation' %}
```

Comment:
Should this be made into a macro - looks like it repeats a few times in this source file